crates/matrix-sdk-crypto/src/identities/room_identity_state.rs

```diff
@@ -161,21 +185,59 @@ impl<R: RoomIdentityProvider> RoomIdentityState<R> {
         user_identity: &UserIdentity,
     ) -> Option<IdentityStatusChange> {
         if let UserIdentity::Other(_) = &user_identity {
-            // If the user's state has changed
-            let new_state = state_of(user_identity);
-            let old_state = self.known_states.get(user_id);
-            if new_state != old_state {
-                Some(self.set_state(user_identity.user_id(), new_state))
-            } else {
-                // Nothing changed
-                None
-            }
+            self.update_user_state_to(user_id, self.room.state_of(user_identity))
         } else {
             // Ignore updates to our own identity
             None
         }
     }
 
+    // If the supplied `new_state` represents an actual change, updates our internal
+    // state for this user, and returns the change information we will surface to
+    // the UI.
```

Comment:
this maybe needs clarification now that there are some cases that just set state, and some that also return an update.

also, /// for doc-comments.

Reply:
Gack, auto-merged. Fixing in a new PR

Reply:
https://github.com/matrix-org/matrix-rust-sdk/pull/4167